third-party/hwloc/Makefile.include

```diff
@@ -6,3 +6,14 @@ HWLOC_INSTALL_SUBDIR=install/$(CHPL_MAKE_HWLOC_UNIQ_CFG_PATH)
 HWLOC_INSTALL_DIR=$(HWLOC_DIR)/$(HWLOC_INSTALL_SUBDIR)
 HWLOC_INCLUDE_DIR=$(HWLOC_INSTALL_DIR)/include
 HWLOC_LIB_DIR=$(HWLOC_INSTALL_DIR)/lib
+
+
+# For consistency with the rest of the runtime code base CHPL_HWLOC_PREFIX ends with "hwloc_".
```

Comment:
I may be missing something -- is it not possible just to define `CHPL_HWLOC_PREFIX=chpl_` (or nothing) and throw `--with-hwloc-symbol-prefix=$CHPL_HWLOC_PREFIX` to both the hwloc and qthreads `configure` scripts? Then default the hwloc prefix string to nothing and do `AC_SEARCH_LIBS([${with_hwloc_symbol_prefix}hwloc_topology_init], ...`in qthread_check_hwloc.m4?

Reply:
I think we want CHPL_HWLOC_PREFIX to be "chpl_hwloc_" to be consistent with the value of CHPL_JEMALLOC_PREFIX (chpl_je_). The problem is if you that value of CHPL_HWLOC_PREFIX to  --with-hwloc-symbol-prefix for hwloc you end up with the prefix being "chpl_hwloc_hwloc_" (this is essentially a bug on their part because it doesn't work the way it's documented or the way other packages handle prefixes). You can't pass the same prefix to hwloc and qthread, so I hid the ugliness in the hwloc Makefile.

Reply:
I'd defer to @ronawho on this one, too, but personally I don't think we have to try to make the hwloc prefix look like it works the same as the jemalloc one even though it doesn't. In fact, making it more obvious that it's different seems better, to me. Since Qthreads hasn't previously had support for a hwloc prefix, aren't we in control of whether the hwloc symbol prefix specified to the Qthreads configure script needs to include the `hwloc_` part or not? It seems as though we could just make the Qthreads behavior with respect to the hwloc prefix match the hwloc behavior, even though neither matches the jemalloc behavior.